storage/innobase/dict/dict0stats.cc

```diff
@@ -3005,17 +2977,19 @@ dict_stats_save(
 		"END;", trx);
 
 	if (UNIV_UNLIKELY(ret != DB_SUCCESS)) {
-		ib::error() << "Cannot save table statistics for table "
-			<< table->name << ": " << ret;
+		sql_print_error("InnoDB: Cannot save table statistics for"
+				" table %`.*s.%`s: %s",
+				int(table->name.dblen()), table->name.m_name,
+				table->name.basename(), ut_strerr(ret));
```

Comment:
This change turns out to have caused a crash in the following test:
```sh
./mtr --embedded innodb_fts.misc_debug,orig
```
The cause is that the implementation of `vprint_msg_to_log()` in `libmysqld/lib_sql.cc` invokes the standard library function `vsnprintf()`, which cannot deal with the backtick modifier. The result is a crash when something is trying to compute `strlen()` on the length 4 (the length of the first component of `test/t1`). That is, the first parameter of the format is being misinterpreted as a pointer instead of a length.

I figured out that this can be checked by `#ifdef EMBEDDED_SERVER`.